src/sentry/tsdb/snuba.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.tsdb.base import BaseTSDB, TSDBModel
+from sentry.utils import snuba
+
+
+class SnubaTSDB(BaseTSDB):
+    """
+    A time series query interface to Snuba
+
+    Write methods are not supported, as the raw data from which we generate our
+    time series is assumed to already exist in snuba.
+
+    Read methods are supported only for models based on group/event data and
+    will return empty results for unsupported models.
+    """
+
+    def __init__(self, **options):
+        super(SnubaTSDB, self).__init__(**options)
+
+    def model_columns(self, model):
+        """
+        Translates TSDB models into the required columns for querying snuba.
+        Returns a tuple of (groupby_column, aggregateby_column)
+        """
+        return {
+            TSDBModel.project: ('project_id', None),
+            TSDBModel.group: ('issue', None),
+            TSDBModel.release: ('release', None),
+            TSDBModel.users_affected_by_group: ('issue', 'user_id'),
+            TSDBModel.users_affected_by_project: ('project_id', 'user_id'),
+            TSDBModel.users_affected_by_project: ('project_id', 'user_id'),
+            TSDBModel.frequent_environments_by_group: ('issue', 'environment'),
+            TSDBModel.frequent_releases_by_group: ('issue', 'release'),
+            TSDBModel.frequent_issues_by_project: ('project_id', 'issue'),
+        }.get(model, None)
+
+    def get_data(self, model, keys, start, end, rollup=None, environment_id=None,
+                 aggregation='count', group_on_model=True, group_on_time=False):
+        """
+        Normalizes all the TSDB parameters and sends a query to snuba.
+
+        `group_on_time`: whether to add a GROUP BY clause on the 'time' field.
+        `group_on_model`: whether to add a GROUP BY clause on the primary model.
+        """
+        model_columns = self.model_columns(model)
+
+        if model_columns is None:
+            return None
```

Comment:
This seems like it should raise an exception here if the backend can't handle the request, rather then returning `None`. Otherwise this is probably going to manifest itself as a weird combination of `{Attribute,Type,Value}Error`.